docs/technical-documentation/migrate-csv.md

```diff
@@ -1 +1,815 @@
+
+### Running the media migration
+
+Run `drush -y --uri=http://localhost:8000 migrate:import media` from anywhere within the Drupal installation directory. You should now be able to see the media files attached to the nodes you created earlier. At this point, you might want to create Service Files and Thumbnails using the appropriate Drupal actions on the main content admin window.
+
+## What have we learned?
+
+If you've made it all the way to the end here, then you've learned that you can migrate files and CSV metadata into Islandora using only yml files.  You've seen how to transform data with pipelines of processing plugins and can handle numeric, text, and entity reference fields.  You can handle multiple values for fields, and even more complicated things like `typed_relation` fields.  And as big as this walkthrough was, we're only scratching the surface of what can be done with the Migrate API.
+
+## Where to go from here?
+
+There's certainly more you can do with Drupal 8's Migrate API.  There's a plethora of source and processing plugins out there that can handle pretty much anything you throw at it.  XML and JSON are fair game.  You can also request sources using HTTP, so you can always point it at an existing systems REST API and go from there.  If you can't make the Migrate API's existing workflow make the necessary changes to your data, you can expand its capabilities by writing your own process plugins. Reading the Drupal.org documentation on the [Migrate API](https://www.drupal.org/docs/8/api/migrate-api) would be a good place to start.
+
+But really the best thing to do is try and get your data into Islandora!  We intend to create a `boilerplate` branch of this repository that will allow you to clone down a migration template, ready for you to customize to fit your data.  And as you assemble it into CSV format, keep in mind that if you have more than just names for things like subjects and authors, that you can always make more CSVs.  Think of it like maintaining tables in an SQL database.  Each CSV has unique keys, so you can lookup/join entities between CSVs using those keys.  And you can still pipe-delimit the keys like we did in our example to handle multi-valued fields.
+
+In some repositories, these CSVs can be used to make bulk updates to metadata. Just make your changes to maintain the CSVs, then run the migration(s) again with the --update flag. This will not always be efficient, as you'll update every entity, even if it didn't change. But, by breaking down your CSVs per collection or object type, you could keep them small enough to use this process for a small repository.
+
+There is also a tool for migrating directly from an Islandora 7.x to Islandora 8 ([migrate_7x_claw](https://github.com/Islandora-Devops/migrate_7x_claw)), using Solr, XPaths, and Fedora calls to pull files and objects directly into Islandora 8. It may be worth checking out, and/or using in conjunction with a CSV migration.
```

Comment:
Thank you, Seth. The document is definitely well written. It may be too "technical" for me to review. Just one suggestion. Some terms may need their references if they are not in "Islandora glossary". Here are some examples: 
Linked Agent --Line 34 and 682; UI--Line 47; Typed Data AP--Line 588; DSID-Line 780; Service Files-Line 801.
Thanks from Ian

Reply:
No worries. *Disclaimer* I mostly copy/pasted this from the existing tutorial in the referenced module; so, yeah, @dannylamb did a good job on this.

I'll see if I can link some of those terms. Some are documented elsewhere in the docs; although DSID is an Islandora 7 thing that I'm not sure where to link to.

Reply:
Hopefully some of those edits will make it more clear.